compiler/codegen/symbol.cpp

```diff
@@ -1831,6 +1851,88 @@ GenRet FnSymbol::codegen() {
   return ret;
 }
 
+void FnSymbol::codegenFortran(int indent) {
+  GenInfo *info = gGenInfo;
+  int beginIndent = indent;
+
+  if (!hasFlag(FLAG_EXPORT)) return;
+  if (hasFlag(FLAG_NO_PROTOTYPE)) return;
+  if (hasFlag(FLAG_NO_CODEGEN)) return;
+
+  if (info->cfile) {
+    FILE* outfile = info->cfile;
+    if (fGenIDS)
+      fprintf(outfile, "%*s! %s", indent, "", idCommentTemp(this));
```

Comment:
I don't understand why you're calling idCommentTemp as it generates C++ comments.
I see you have the `!` here but I think it'd be simpler to just have
`fprintf(outfile, "%*s! %i", indent, "", this->id);`